src/amplify-migration/amplifyMigrationScript.ts

```diff
@@ -0,0 +1,289 @@
+}
+
+const listOfRepos: [string, string][] = [
+  ["repo-name", "Human Friendly Name"], // Modify this line when running the script
+];
+
+async function main() {
+  for (const [repoName, name] of listOfRepos) {
+    try {
+      await migrateRepo(repoName, name);
+    } catch (e) {
+      console.error(e);
+    }
+  }
+}
+
+async function migrateRepo(repoName: string, name: string) {
+  const homeDir = process.env.HOME;
+  const repoPath = `${homeDir}/isomer-migrations/${repoName}`;
+
+  try {
+    const build_spec = await readBuildSpec();
+    const appId = await createAmplifyApp(repoName, build_spec);
+    const amplifyAppInfo: AmplifyAppInfo = {
+      appId,
+      repoName,
+      name,
+      repoPath,
+    };
+    await createAmplifyBranches(amplifyAppInfo);
+    await startReleaseJob(amplifyAppInfo);
+    await modifyRepo(amplifyAppInfo);
+    await pushChangesToRemote(amplifyAppInfo);
+    await generateSqlCommands(amplifyAppInfo);
+  } catch (e) {
+    console.error(e);
+  }
+}
+
+async function modifyRepo({ repoName, appId, repoPath }: AmplifyAppInfo) {
+  if (fs.existsSync(repoPath)) {
+    console.log(
+      `Repository ${repoName} already exists. Pulling changes from origin.`
+    );
+    await simpleGit(repoPath).pull("origin", "staging");
+  } else {
+    console.log(`Cloning ${repoName} repository from ${ORGANIZATION_NAME}...`);
+    await simpleGit().clone(
+      `https://github.com/${ORGANIZATION_NAME}/${repoName}.git`,
+      repoPath
+    );
+  }
+
+  // get list of branches
+  const branches = await simpleGit(repoPath).branchLocal();
+  if (branches.all.includes(BRANCH_NAME)) {
+    console.log("Branch already exists. Checking out branch.");
+    await simpleGit(repoPath).checkout(BRANCH_NAME);
+  } else {
+    await simpleGit(repoPath).checkoutLocalBranch(BRANCH_NAME);
+  }
+
+  await modifyPermalinks(repoPath);
+
+  await updateConfigYml(appId, repoPath);
+}
+
+async function modifyPermalinks(repoPath: string) {
+  const mdFiles = await new Promise((resolve, reject) => {
+    glob("**/*.md", { cwd: repoPath }, (err: any, files: any) => {
+      if (err) {
+        reject(err);
+        throw err;
+      } else {
+        resolve(files);
+      }
+    });
+  });
+
+  //dictionary  of changed permalinks
+  const changedPermalinks: { [key: string]: string } = {};
+
+  // eslint-disable-next-line no-restricted-syntax
+  for (const file of Object.values(mdFiles as {})) {
+    const filePath = path.join(repoPath, file);
+    const fileContent = (await fs.promises.readFile(filePath)).toString();
+    const permalinkRegex = /^permalink: /m;
+    const permalinkIndex = fileContent.search(permalinkRegex);
+
+    if (permalinkIndex !== -1) {
+      const permalinkLine = fileContent.slice(
+        permalinkIndex,
+        fileContent.indexOf("\n", permalinkIndex)
+      );
+      let permalinkLineTrimmed = permalinkLine.trim();
+
+      let hasTrailingQuotation = false;
+      if (permalinkLineTrimmed.endsWith(`"`)) {
```

Comment:
I would keep this condition to err on the side of caution since: 
1) we might need to migrate websites that are pre CMS 
2) we dont have a guarantee that we disallowed `"` for **_all_** CMS websites 
3) trying to find such an edge case that exists in step 2 takes more effort than adding this condition 
4) (correct me if I am wrong here) there should not be any negative side effects of this condition

Reply:
fair aa9302f